include/range/v3/range_concepts.hpp

```diff
@@ -141,6 +142,25 @@ namespace ranges
                     ));
             };
 
+            struct ContiguousRange
+              : refines<RandomAccessRange>
+            {
+                template<typename Rng>
+                using data_reference_t = decltype(*ranges::data(std::declval<Rng&>()));
+
+                template<typename Rng>
+                using datum_t = meta::_t<std::remove_reference<data_reference_t<Rng>>>;
+
+                template<typename Rng>
+                auto requires_(Rng && rng) -> decltype(
+                    concepts::valid_expr(
+                        concepts::model_of<Same, InputRange::value_t<Rng>,
+                            meta::_t<std::remove_cv<datum_t<Rng>>>>(),
+                        concepts::model_of<CommonReference, InputRange::reference_t<Rng>,
+                            data_reference_t<Rng>>()
+                    ));
+            };
```

Comment:
Contiguity of iterators is (as of C++17) purely semantic; given an iterator type, there's no way to determine if that iterator is a contiguous iterator. Given a range type, I can determine syntactically if `data` is defined for that range and returns a pointer to an object type that is cv-compatible with its iterator's reference type. (I strongly suspect `CommonReference` is the wrong requirement for this.) The presence of such a `data` member strongly correlates with contiguity of ranges, at least in the standard library.